### ISSUE TITLE
ability to filter when playing back

### ISSUE THREAD

**Issue by mvernimmen** [buger/goreplay#369]
Currently the --http-disallow and -http-allow filters seem to only work when recording traffic. I would like te record all traffic but when playing back, only play back those things matched by the filter. Possibly I could use middleware to do this, but that would be extra work for something that already seems to exist in gor.
Is filtering on playback something you could add?

**Comment by buger** [comment]
Do you mean that this filters did not worked for you when using --input-file?
Can you provide the full command which you used?

`--input-file` should work fine with those filters by design, if it's not, maybe you hit some bug.

Thank you!

**Comment by mvernimmen** [comment]
This is what I tried:
```
[m.vernimmen@md001 ~]$ ./gor --input-file "PROD-diamond_22.log" --output-stdout --http-disallow-header 'Host: www.ver.*' | grep 'Host: www.ver'
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
^C
[m.vernimmen@md001 ~]$ ./gor --input-file "PROD-diamond_22.log" --output-stdout --http-disallow-header 'Host: www.vertaa.fi' | grep 'Host: www.ver'
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
Host: www.vertaa.fi
^C
[m.vernimmen@md001 ~]$ ./gor --input-file "PROD-diamond_22.log" --output-stdout --http-disallow-url '^www.vertaa.fi' | grep 'www.vertaa.fi'
Host: www.vertaa.fi
Referer: https://www.vertaa.fi/digikamerat/samsung/
^C
[m.vernimmen@md001 ~]$ ./gor --input-file "PROD-diamond_22.log" --output-stdout --http-disallow-url 'www.vertaa.fi' | grep 'www.vertaa.fi'
Host: www.vertaa.fi
Referer: https://www.vertaa.fi/digikamerat/samsung/
^C
[m.vernimmen@md001 ~]$
```

**Comment by buger** [comment]
Sorry that it took so long to fix, here is fix #412 and fresh binary:
[gor_DEV-1482853957_x64.tar.gz](https://github.com/buger/goreplay/files/674330/gor_DEV-1482853957_x64.tar.gz)
